### PR TITLE
Correcting String Internal field names

### PR DESCRIPTION
**Pull request by priyank23**
@e-kayrakli I have tried to correct the internal field names of String module as stated in the recent issue. I would love to have some review over the changes and please suggest any other modifications required.

### CODE REVIEW

**Comment by Rapiz1** [comment]
1. You need a [CLA](https://github.com/chapel-lang/chapel/blob/e90f0b518d8fce5273acf5467fc2f7e182bad2eb/.github/CONTRIBUTING.md) before contributing to Chapel
2. CI failed to build your branch. You need to check out the log and fix it.

**Comment by ankingcodes** [comment]
```
$CHPL_HOME/modules/standard/HaltWrappers.chpl:137: In function 'safeCastCheckHalt':
$CHPL_HOME/modules/standard/HaltWrappers.chpl:138: error: unresolved call 'string.isowned'
$CHPL_HOME/modules/internal/Bytes.chpl:339: note: this candidate did not match: bytes.isowned
$CHPL_HOME/modules/internal/BytesStringCommon.chpl:188: note: because method call receiver with type string
$CHPL_HOME/modules/internal/Bytes.chpl:339: note: is passed to formal 'ref this: bytes'
```
@priyank23  The refactor is breaking these modules.

**Comment by priyank23** [comment]
@ankingcodes Thanks for the assist.

**Comment by priyank23** [comment]
@Rapiz1 Sure, Will submit a CLA as soon as possible.

**Comment by e-kayrakli** [comment]
@priyank23 -- I think this looks good. Potential errors on PRs like this can only be found through running a lot of tests. Are you familiar with our testing system? If not check out [test system documentation](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst). Luckily, you only need to know about "Invoking start_test" to run existing tests. So don't be scared :) I think being able to run `start_test` and observe the results is an essential part of contributing to Chapel (GSoC or otherwise).

As a quick summary you should
- Run `make test-venv` in Chapel root directory
- See if you can run `start_test` on master.
  Checkout master branch and not this branch. Then see if you can `start_test test/release/examples/hello.chpl` to see if the hello world test runs successfully.
- Now for your branch, I'd definitely test `test/types/string/kbrady` and `test/release/examples/primers`
- If both of those are successful, I'd be up for running the full testing suite on this. (It takes too long and we have parallel systems to run it quicker) I can let you know any other follow up issues.
  - I am particularly concerned because there may be some internal code that plays with these fields directly. Which would also need to be fixed.
  - Some of that code might be in `modules/internal/BytesStringCommon.chpl` which also implies that this refactor is not possible without changing the fields of the `bytes` type, too. Which would not be a bad thing, at all. See `modules/internal/Bytes.chpl`.

**Comment by priyank23** [comment]
@e-kayrakli I have ran the above mentioned tests successfully on my system. Also, I have modified the analysed the internal code(probably all of them) using these fields directly. Please review and suggest the changes, if required.

**Comment by e-kayrakli** [comment]
Thanks! All that sounds good. On a quick look, I have seen that you have made some changes in the Regexp module, too. Now, for the internal consistency, it may be a good idea, but it may be not. What I was suggesting that in some internal modules we may have code like:

```chapel
var s: string;
s.buf = blah;
s.len = 100;
```
Such uses of fields should also change for correctness purposes. But what you have done seems more like changing the field of an extern record. I wouldn't go that far in this PR.

And now that you changed some more code, maybe you can also test `start_test test/types/bytes`

Now for the procedure, as you may already know we are churning towards our next release, so things are a bit busy. And, as inconsequential as this changes may be, they are still large and touches internal parts of the language. So, I will add a "post-release" label to this PR and will hold off merging until we cut our release from the master branch. And until that happens, I will checkout this branch and test it locally and against our full testing suite. I will let you know of any failures. It'll take a little while but this'll get merged eventually.

[Chapel release process](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/ContributorInfo.rst#chapel-release-process) has some more information about this if you don't know already.

**Comment by priyank23** [comment]
@e-kayrakli Sure, I will keep looking for updates by you in this thread.

File: modules/standard/Regexp.chpl [hunk 1]
```diff
@@ -387,7 +387,7 @@ extern const QIO_REGEXP_ANCHOR_BOTH:c_int;
 pragma "no doc"
 extern record qio_regexp_string_piece_t {
   var offset:int(64); // counting from 0, -1 means "NULL"
-  var len:int(64);
+  var buffLen:int(64);
```
Comment (e-kayrakli) [inline modules/standard/Regexp.chpl]: let's not change this here

Comment (priyank23) [reply modules/standard/Regexp.chpl]: I have done the changes requested. I also ran tests ```start_test test/types/string/kbrady``` ```start_test test/release/examples/primers``` and ```start_test test/types/bytes``` successfully in my system. Kindly review and notify me of any more changes required.

Comment (priyank23) [reply modules/standard/Regexp.chpl]: Also, I accidently deleted the ```rst/``` a couple of days back, hence, the large changes that are shown [here](https://github.com/chapel-lang/chapel/pull/15225/commits/b289ca64cc608cae0b3a62b7924bc96c051f35ee). In the mentioned commit, only ```Regexp.chpl``` file has been modified. Kindly ignore the others. Thank you

**Comment by e-kayrakli** [comment]
There are two things I'll need before being able to run testing on this:

- Revert back your changes w.r.t rst files. I think you have added a dir 'rst' in the root and added that to the repo. That's not right. Some of the rst files we have are autogenerated through chpldoc and shouldn't be in the repo. You'll need to `git rm -f rst` and `git checkout master -- doc` to revert those files back. I also see some binaries added, that shouldn't happen either. You can run a similar `git checkout` command for that.

In the end, you can do a quick sanity check by running `git diff --stat master` to see what is changed in your feature branch compared to master, assuming that master hasn't changed since you branched off. If you updated master, you can get the base commit while on this branch by `git merge-base HEAD master` then do `git diff --stat <commit SHA from previous command>`

- You should fix conflicts. Some (all?) of them are introduced by my recent PR https://github.com/chapel-lang/chapel/pull/15386, please make sure you don't remove those fixes.

To resolve conflicts, you should update your master by running`git pull upstream master` on master branch (replace upstream with whatever you have for the main Chapel repo, you can see your remotes and their links via `git remote -v`), then checkout this branch and do `git merge master`. It'd be a good practice to try `git rebase master` instead but it may get a bit trickier.

p.s. feel free to ignore my `git` recommendations if you are fluent in `git` and do things your way

**Comment by priyank23** [comment]
Due to some unresolvable issues, I had to close this PR. The corresponding PR created is #15490 . Kindly, refer to that

**Pull request closed without merging**